problem2.cpp

```diff
@@ -0,0 +1,219 @@
+                            break;
+                        }
+                        else if(temp.str.at(i)<res.str.at(i)){
+                            res.str = temp.str;
+                            break;
+                        }
+                    }
+                }
+            }
+            //cout<<res.str<<" ";
+            //cout<<res.len<<" "<<endl;
+        }
+        return res.str;
+    }
+};
+
+// attempt - 2
+
+class Solution {
+public:
+    string longestWord(vector<string>& words) {
+        TrieNode* root = new TrieNode();
+        vector<int> hash_len(26,0);
+        vector<string> hash_str(26,"");
+        int m_len{0};
+        string m_str{""};
+        for(string s: words){
+            int hash_id = s.at(0) - 'a';
+            TrieNode* obj = root;
+            for(char c:s){
+                int idx = c - 'a';
+                if(obj->child.at(idx) == nullptr){
+                    obj->child.at(idx) = new TrieNode();
+                }
+                obj = obj->child.at(idx);
+            }
+            //obj->flag = true;
+            hash_len[hash_id]++;
+            //below code is to store the maximum length string for the starting alphabet
+            int m = s.length();
+            int n = hash_str[hash_id].length();
+            if(m>n){
+                hash_str[hash_id] = s;
+            }
+            else if(m == n){
+                for(int i{};i<n;++i){
+                    //out<<temp.str.at(i)<<" "<<res.str.at(i)<<"\t";
+                    if(s.at(i)>hash_str[hash_id].at(i)){
+                        break;
+                    }
+                    else if(s.at(i)<hash_str[hash_id].at(i)){
+                        hash_str[hash_id] = s;
+                        break;
+                    }
+                }
+            }
+            // here we try to find the one with maximum level
+            // if level are same then we see if they are lexicographically big or small
+            if(hash_len[hash_id]>m_len){
+                m_len = hash_len[hash_id];
+                m_str = hash_str[hash_id];
+            }
+            else if(hash_len[hash_id] == m_len){
+                int l1 = hash_str[hash_id].length();
+                int l2 = m_str.length();
+                
+                if(l1>l2){
+                    m_str = hash_str[hash_id];
+                }
+                else if(l1 == l2){
+                    for(int i{};i<l1;++i){
+                        if(hash_str[hash_id].at(i)>m_str.at(i)){
+                            break;
+                        }
+                        else if(hash_str[hash_id].at(i)<m_str.at(i)){
+                            m_str = hash_str[hash_id];
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        return m_str;
+    }
+};
+
+//above attemps are wrong becasue didnt consider one character at a time.
+
+//attempt - 3 google
```

Comment:
I would say if you are not able to solve a problem before it's discussed in class wait for the class instead of googling the solution